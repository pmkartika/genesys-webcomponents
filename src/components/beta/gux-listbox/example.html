<h1>gux-listbox</h1>
<form onchange="notify(event)" oninput="notify(event)">
  <h3>Typical</h3>
  <gux-listbox tabindex="0" value="b" aria-label="Animals">
    <gux-option-v2 value="a" disabled>Ant</gux-option-v2>
    <gux-option-v2 value="b">Bat</gux-option-v2>
    <gux-option-v2 value="c">Cat</gux-option-v2>
    <gux-option-v2 value="d">Dog</gux-option-v2>
    <gux-option-v2 value="e" disabled>Eel</gux-option-v2>
    <gux-option-v2 value="f">Frog</gux-option-v2>
    <gux-option-v2 value="g">Goat</gux-option-v2>
    <gux-option-v2 value="h" disabled>Horse</gux-option-v2>
    <gux-option-v2 value="i" disabled>Ibis</gux-option-v2>
  </gux-listbox>

  <h3>Disabled</h3>
  <gux-listbox tabindex="0" value="a" aria-label="Animals">
    <gux-option-v2 value="a" disabled>Ant</gux-option-v2>
    <gux-option-v2 value="b">Bat</gux-option-v2>
    <gux-option-v2 value="c">Cat</gux-option-v2>
    <gux-option-v2 value="d">Dog</gux-option-v2>
    <gux-option-v2 value="e" disabled>Eel</gux-option-v2>
    <gux-option-v2 value="f">Frog</gux-option-v2>
    <gux-option-v2 value="g">Goat</gux-option-v2>
    <gux-option-v2 value="h" disabled>Horse</gux-option-v2>
    <gux-option-v2 value="i" disabled>Ibis</gux-option-v2>
  </gux-listbox>

  <h3>Ids</h3>
  <gux-listbox tabindex="0" aria-label="Animals">
    <gux-option-v2 id="ant" value="a">Ant</gux-option-v2>
    <gux-option-v2 id="bat" value="b">Bat</gux-option-v2>
    <gux-option-v2 id="cat" value="c">Cat</gux-option-v2>
    <gux-option-v2 id="dog" value="d">Dog</gux-option-v2>
  </gux-listbox>
</form>
