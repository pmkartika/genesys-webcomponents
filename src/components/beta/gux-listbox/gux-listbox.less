@import (reference) '../../../style/color.less';
@import (reference) '../../../style/mixins.less';
@import (reference) '../../../style/shadows.less';

:host {
  box-sizing: border-box;
  display: block;
  max-height: 20rem;
  padding: 8px 0;
  margin: 0;
  overflow-y: auto;
  color: @gux-black-50;
  background: @gux-grey-100;
  border: 1px solid @gux-grey-30;
  border-radius: 4px;
  outline: none;
}

:host(:focus) {
  .focus-ring();
}

.gux-no-matches {
  box-sizing: border-box;
  height: 32px;
  padding: 8px 16px;
  color: @gux-black-50;
}
