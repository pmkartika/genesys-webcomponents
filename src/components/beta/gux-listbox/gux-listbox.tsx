import {
  Component,
  Element,
  Event,
  EventEmitter,
  h,
  Host,
  JSX,
  Listen,
  Method,
  Prop,
  State
} from '@stencil/core';

import {
  actOnActiveOption,
  clearActiveOptions,
  goToOption,
  hasPreviousOption,
  hasNextOption,
  onClickedOption,
  setFirstOptionActive,
  setInitialActiveOption,
  setLastOptionActive,
  setNextOptionActive,
  setPreviousOptionActive
} from './gux-listbox.service';

import { buildI18nForComponent, GetI18nValue } from '../../../i18n';
import { whenEventIsFrom } from '../../../utils/dom/when-event-is-from';
import simulateNativeEvent from '../../../utils/dom/simulate-native-event';
import { trackComponent } from '../../../usage-tracking';
import { logError } from '../../../utils/error/log-error';

import translationResources from './i18n/en.json';

/**
 * @slot - collection of gux-option-v2s
 */
@Component({
  styleUrl: 'gux-listbox.less',
  tag: 'gux-listbox',
  shadow: true
})
export class GuxListbox {
  private i18n: GetI18nValue;

  @Element()
  root: HTMLGuxListboxElement;

  @Prop({ mutable: true })
  value: string;

  @Prop()
  filter: string = '';

  @State()
  listboxOptions: HTMLGuxOptionV2Element[] = [];

  @State()
  allListboxOptionsFiltered: boolean;

  @Event()
  internallistboxoptionsupdated: EventEmitter;

  @Listen('focus')
  onFocus(): void {
    setInitialActiveOption(this.root);
  }

  @Listen('blur')
  onBlur(): void {
    clearActiveOptions(this.root);
  }

  @Listen('keydown')
  onKeydown(event: KeyboardEvent): void {
    switch (event.key) {
      case 'Enter':
        event.preventDefault();
        actOnActiveOption(this.root, value => this.updateValue(value));
        return;

      case 'ArrowDown':
        event.preventDefault();
        if (hasNextOption(this.root)) {
          event.stopPropagation();
          setNextOptionActive(this.root);
        } else {
          setFirstOptionActive(this.root);
        }
        return;

      case 'ArrowUp': {
        event.preventDefault();
        if (hasPreviousOption(this.root)) {
          event.stopPropagation();
          setPreviousOptionActive(this.root);
        } else {
          setLastOptionActive(this.root);
        }
        return;
      }

      case 'Home': {
        event.preventDefault();
        setFirstOptionActive(this.root);
        return;
      }

      case 'End': {
        event.preventDefault();
        setLastOptionActive(this.root);
        return;
      }

      case ' ': {
        event.preventDefault();
        return;
      }
    }

    if (event.key.length === 1) {
      goToOption(this.root, event.key);
      return;
    }
  }

  @Listen('keyup')
  onKeyup(event: KeyboardEvent): void {
    switch (event.key) {
      case ' ':
        actOnActiveOption(this.root, value => this.updateValue(value));
        return;
    }
  }

  @Listen('mousemove', {})
  onMousemove(): void {
    clearActiveOptions(this.root);
  }

  @Listen('click')
  onClick(event: MouseEvent): void {
    whenEventIsFrom(
      'gux-option-v2',
      event,
      (option: HTMLGuxOptionV2Element) => {
        onClickedOption(option, value => this.updateValue(value));
      }
    );
  }

  // eslint-disable-next-line @typescript-eslint/require-await
  @Method()
  async guxSelectActive(): Promise<void> {
    actOnActiveOption(this.root, value => this.updateValue(value));
  }

  private setListboxOptions(): void {
    this.listboxOptions = Array.from(
      this.root.children
    ) as HTMLGuxOptionV2Element[];
    this.internallistboxoptionsupdated.emit();
  }

  private updateValue(newValue: string): void {
    if (this.value !== newValue) {
      this.value = newValue;

      simulateNativeEvent(this.root, 'input');
      simulateNativeEvent(this.root, 'change');
    }
  }

  async componentWillLoad(): Promise<void> {
    trackComponent(this.root);
    this.i18n = await buildI18nForComponent(this.root, translationResources);

    this.setListboxOptions();
  }

  componentDidLoad(): void {
    if (
      !(
        this.root.getAttribute('aria-label') ||
        this.root.getAttribute('aria-labelledby')
      )
    ) {
      logError(
        'gux-listbox',
        '`gux-listbox` requires a label. Either provide a label and associate it with the listbox using `aria-labelledby` or add an `aria-label` attribute to the gux-listbox element.'
      );
    }
  }

  componentWillRender(): void {
    this.listboxOptions.forEach(listboxOption => {
      listboxOption.selected = listboxOption.value === this.value;
      listboxOption.filtered = !listboxOption.textContent
        .toLowerCase()
        .startsWith(this.filter.toLowerCase());
    });

    this.allListboxOptionsFiltered =
      this.listboxOptions.filter(listboxOption => !listboxOption.filtered)
        .length === 0;

    if (this.filter) {
      setFirstOptionActive(this.root);
    }
  }

  renderAllListboxOptionsFiltered(): JSX.Element {
    if (this.allListboxOptionsFiltered) {
      return (
        <div class="gux-no-matches">{this.i18n('noMatches')}</div>
      ) as JSX.Element;
    }
  }

  render(): JSX.Element {
    return (
      <Host role="listbox" tabindex={0}>
        <slot onSlotchange={() => this.setListboxOptions()} />
        {this.renderAllListboxOptionsFiltered()}
      </Host>
    ) as JSX.Element;
  }
}
