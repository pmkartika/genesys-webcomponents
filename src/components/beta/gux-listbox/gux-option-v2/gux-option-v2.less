@import (reference) '../../../../style/color.less';

gux-option-v2 {
  box-sizing: border-box;
  display: flex;
  min-height: 32px;
  padding: 8px 16px;
  color: @gux-black-50;
  word-wrap: break-word;
  cursor: pointer;

  &.gux-disabled {
    pointer-events: none;
    cursor: default;
    opacity: 0.5;
  }

  &.gux-selected {
    background: @gux-blue-100;
  }

  &.gux-active,
  &.gux-hovered:not([disabled]) {
    color: @gux-grey-100;
    background: @gux-blue-60;
  }

  &.gux-filtered {
    display: none;
  }
}
