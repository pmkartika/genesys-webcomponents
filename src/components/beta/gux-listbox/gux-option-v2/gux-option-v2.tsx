import { Component, Element, h, Host, JSX, Listen, Prop } from '@stencil/core';

import { randomHTMLId } from '../../../../utils/dom/random-html-id';

@Component({
  styleUrl: 'gux-option-v2.less',
  tag: 'gux-option-v2'
})
export class GuxOptionV2 {
  @Element()
  root: HTMLElement;

  @Prop()
  value: string;

  @Prop()
  active: boolean = false;

  @Prop()
  selected: boolean = false;

  @Prop()
  disabled: boolean = false;

  @Prop()
  filtered: boolean = false;

  @Prop({ mutable: true })
  hovered: boolean = false;

  @Listen('mouseenter')
  onmouseenter() {
    this.hovered = true;
  }

  @Listen('mouseleave')
  onMouseleave() {
    this.hovered = false;
  }

  componentWillLoad(): void {
    this.root.id = this.root.id || randomHTMLId('gux-option-v2');
  }

  private getAriaSelected(): boolean | string {
    if (this.disabled) {
      return false;
    }

    return this.selected ? 'true' : 'false';
  }

  render(): JSX.Element {
    return (
      <Host
        role="option"
        class={{
          'gux-active': this.active,
          'gux-disabled': this.disabled,
          'gux-filtered': this.filtered,
          'gux-hovered': this.hovered,
          'gux-selected': this.selected
        }}
        aria-selected={this.getAriaSelected()}
        aria-disabled={this.disabled.toString()}
      >
        <slot />
      </Host>
    ) as JSX.Element;
  }
}
