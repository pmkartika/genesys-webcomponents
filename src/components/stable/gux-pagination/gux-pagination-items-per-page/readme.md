# gux-pagination-items-per-page

An internal component used by the gux-pagination component.


<!-- Auto Generated Below -->


## Properties

| Property       | Attribute        | Description | Type                    | Default |
| -------------- | ---------------- | ----------- | ----------------------- | ------- |
| `itemsPerPage` | `items-per-page` |             | `100 \| 25 \| 50 \| 75` | `25`    |


## Events

| Event                        | Description | Type                  |
| ---------------------------- | ----------- | --------------------- |
| `internalitemsperpagechange` |             | `CustomEvent<number>` |


## Dependencies

### Used by

 - [gux-pagination](..)

### Depends on

- [gux-dropdown-v2-beta](../../../beta/gux-dropdown-v2)
- [gux-listbox](../../../beta/gux-listbox)
- [gux-option-v2](../../../beta/gux-listbox/gux-option-v2)

### Graph
```mermaid
graph TD;
  gux-pagination-items-per-page --> gux-dropdown-v2-beta
  gux-pagination-items-per-page --> gux-listbox
  gux-pagination-items-per-page --> gux-option-v2
  gux-dropdown-v2-beta --> gux-icon
  gux-dropdown-v2-beta --> gux-popup-beta
  gux-pagination --> gux-pagination-items-per-page
  style gux-pagination-items-per-page fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
