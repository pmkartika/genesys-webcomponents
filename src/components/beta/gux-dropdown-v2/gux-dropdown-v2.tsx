import {
  Component,
  Element,
  forceUpdate,
  h,
  Listen,
  JSX,
  Prop,
  State,
  Watch
} from '@stencil/core';

import { OnClickOutside } from '../../../utils/decorator/on-click-outside';
import { buildI18nForComponent, GetI18nValue } from '../../../i18n';
import simulateNativeEvent from '../../../utils/dom/simulate-native-event';
import { trackComponent } from '../../../usage-tracking';

import translationResources from './i18n/en.json';

import { getSearchOption } from '../gux-listbox/gux-listbox.service';

/**
 * @slot - for gux-list-box
 */
@Component({
  styleUrl: 'gux-dropdown-v2.less',
  tag: 'gux-dropdown-v2-beta',
  shadow: true
})
export class GuxDropdownV2Beta {
  private i18n: GetI18nValue;
  private fieldButtonElement: HTMLElement;
  private filterElement: HTMLInputElement;
  private listboxElement: HTMLGuxListboxElement;

  @Element()
  private root: HTMLElement;

  @Prop({ mutable: true })
  value: string;

  @Prop()
  disabled: boolean = false;

  @Prop()
  placeholder: string;

  @Prop()
  filterable: boolean = false;

  @State()
  private expanded: boolean = false;

  @State()
  private filter: string = '';

  @Watch('expanded')
  focusSelectedItemAfterRender(expanded: boolean) {
    if (expanded && this.listboxElement) {
      requestAnimationFrame(() => {
        requestAnimationFrame(() => {
          this.listboxElement.focus();

          if (this.filterable) {
            this.filterElement.focus();
          }
        });
      });
    }

    if (!expanded) {
      this.filter = '';
    }
  }

  @Watch('value')
  validateValue(newValue: string) {
    if (newValue === undefined) {
      this.listboxElement.value = newValue;
      return;
    }

    const selectedListboxOptionElement = this.getOptionElementByValue(newValue);

    if (selectedListboxOptionElement) {
      this.listboxElement.value = newValue;
      return;
    }

    this.value = undefined;
  }

  @Listen('keydown')
  onKeydown(event: KeyboardEvent): void {
    switch (event.key) {
      case 'Escape':
        if (this.filterable) {
          if (document.activeElement === this.listboxElement) {
            return this.filterElement.focus();
          }
        }
        this.collapseListbox('focusFieldButton');
        return;
      case 'Tab':
        if (this.shiftTabFromFilterListbox(event)) {
          event.preventDefault();
          return this.filterElement.focus();
        } else if (this.shiftTabFromExpandedFilterInput(event)) {
          event.preventDefault();
          return this.collapseListbox('focusFieldButton');
        } else {
          this.collapseListbox('noFocusChange');
        }
        return;
      case 'ArrowDown':
        if (this.activeElementNotListbox()) {
          event.preventDefault();
          this.expanded = true;
        }
        return;
    }
  }

  @Listen('internallistboxoptionsupdated')
  onInternallistboxoptionsupdated(event: CustomEvent): void {
    event.stopPropagation();
    forceUpdate(this.root);
  }

  @OnClickOutside({ triggerEvents: 'mousedown' })
  onClickOutside() {
    this.collapseListbox('noFocusChange');
  }

  async componentWillLoad(): Promise<void> {
    trackComponent(this.root);
    this.i18n = await buildI18nForComponent(this.root, translationResources);

    this.listboxElement = this.root.querySelector('gux-listbox');
    this.validateValue(this.value);
  }

  componentDidLoad(): void {
    this.listboxElement.addEventListener('input', (event: InputEvent) => {
      event.stopPropagation();

      this.updateValue((event.target as HTMLGuxListboxElement).value);
    });
    this.listboxElement.addEventListener('change', (event: InputEvent) => {
      event.stopPropagation();
    });
  }

  componentWillRender(): void {
    this.validateValue(this.value);
    this.listboxElement.filter = this.filter;
  }

  private getOptionElementByValue(value: string): HTMLGuxOptionV2Element {
    const listboxOptionElements = this.root.querySelectorAll('gux-option-v2');

    return Array.from(listboxOptionElements).find(
      listboxOptionElement => listboxOptionElement.value === value
    );
  }

  private fieldButtonClick(): void {
    this.expanded = !this.expanded;
  }

  private filterInput(event: InputEvent): void {
    event.stopPropagation();
    this.filter = this.filterElement.value;
  }

  private shiftTabFromExpandedFilterInput(event: KeyboardEvent): boolean {
    return (
      event.shiftKey &&
      this.filterable &&
      this.expanded &&
      !(document.activeElement === this.listboxElement)
    );
  }

  private shiftTabFromFilterListbox(event: KeyboardEvent): boolean {
    return (
      event.shiftKey &&
      this.filterable &&
      document.activeElement === this.listboxElement
    );
  }

  private activeElementNotListbox(): boolean {
    return document.activeElement !== this.listboxElement;
  }

  private filterKeydown(event: KeyboardEvent): void {
    switch (event.key) {
      case 'ArrowDown':
        event.stopImmediatePropagation();
        this.listboxElement.focus();
        return;
      case 'Enter':
        void this.listboxElement.guxSelectActive();
        event.preventDefault();
        return;
    }
  }

  private filterKeyup(event: KeyboardEvent): void {
    switch (event.key) {
      case ' ':
        event.preventDefault();
        return;
    }
  }

  private collapseListbox(
    focusChange: 'noFocusChange' | 'focusFieldButton'
  ): void {
    if (this.expanded) {
      this.expanded = false;
    }

    if (focusChange === 'focusFieldButton') {
      this.fieldButtonElement.focus();
    }
  }

  private updateValue(newValue: string): void {
    if (this.value !== newValue) {
      this.value = newValue;
      this.collapseListbox('focusFieldButton');
      simulateNativeEvent(this.root, 'input');
      simulateNativeEvent(this.root, 'change');
    }
  }

  private getSuggestionText(filter: string): string {
    const filterLength = filter.length;
    if (filterLength > 0) {
      const option = getSearchOption(this.listboxElement, filter);

      if (option) {
        return option.textContent.substring(filterLength);
      }
    }

    return '';
  }

  private renderTargetDisplay(): JSX.Element {
    const selectedListboxOptionElement = this.getOptionElementByValue(
      this.value
    );

    if (selectedListboxOptionElement) {
      return (
        <div class="gux-selected-option">
          {selectedListboxOptionElement.textContent}
        </div>
      ) as JSX.Element;
    }

    return (
      <div class="gux-placeholder">
        {this.placeholder || this.i18n('noSelection')}
      </div>
    ) as JSX.Element;
  }

  private renderFilterInputField(): JSX.Element {
    if (this.expanded && this.filterable) {
      return (
        <div class="gux-field gux-input-field">
          <div class="gux-field-content">
            <div class="gux-filter">
              <div class="gux-filter-display">
                <span class="gux-filter-text">{this.filter}</span>
                <span class="gux-filter-suggestion">
                  {this.getSuggestionText(this.filter)}
                </span>
              </div>
              <div class="input-and-dropdown-button">
                <input
                  onClick={this.fieldButtonClick.bind(this)}
                  class="gux-filter-input"
                  type="text"
                  aria-label={this.i18n('filterResults')}
                  ref={el => (this.filterElement = el)}
                  onInput={this.filterInput.bind(this)}
                  onKeyDown={this.filterKeydown.bind(this)}
                  onKeyUp={this.filterKeyup.bind(this)}
                ></input>
              </div>
            </div>
          </div>
        </div>
      ) as JSX.Element;
    }
  }

  private renderPopup(): JSX.Element {
    return (
      <div slot="popup" class="gux-listbox-container">
        <slot></slot>
      </div>
    ) as JSX.Element;
  }

  private renderTarget(): JSX.Element {
    return (
      <div
        class={{
          'gux-target-container-expanded': this.expanded && this.filterable,
          'gux-target-container-collapsed': !(this.expanded && this.filterable)
        }}
        slot="target"
      >
        {this.renderFilterInputField()}
        <button
          type="button"
          class="gux-field gux-field-button"
          disabled={this.disabled}
          onClick={this.fieldButtonClick.bind(this)}
          ref={el => (this.fieldButtonElement = el)}
          aria-haspopup="listbox"
          aria-expanded={this.expanded.toString()}
        >
          {this.renderTargetContent()}
          <gux-icon
            class={{
              'gux-expand-icon': true
            }}
            screenreader-text={this.i18n('dropdown')}
            iconName="chevron-small-down"
          ></gux-icon>
        </button>
      </div>
    ) as JSX.Element;
  }
  private renderTargetContent(): JSX.Element {
    if (!(this.expanded && this.filterable)) {
      return (
        <div class="gux-field-content">{this.renderTargetDisplay()}</div>
      ) as JSX.Element;
    }
  }

  render(): JSX.Element {
    return (
      <gux-popup-beta expanded={this.expanded} disabled={this.disabled}>
        {this.renderTarget()}
        {this.renderPopup()}
      </gux-popup-beta>
    ) as JSX.Element;
  }
}
