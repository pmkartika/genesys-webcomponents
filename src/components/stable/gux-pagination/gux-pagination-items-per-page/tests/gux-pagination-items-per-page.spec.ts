import { newSpecPage } from '@stencil/core/testing';
import { GuxPaginationItemsPerPage } from '../gux-pagination-items-per-page';
import { GuxDropdownV2Beta } from '../../../../beta/gux-dropdown-v2/gux-dropdown-v2';
import { GuxListbox } from '../../../../beta/gux-listbox/gux-listbox';
import { GuxOptionV2 } from '../../../../beta/gux-listbox/gux-option-v2/gux-option-v2';

const components = [
  GuxPaginationItemsPerPage,
  GuxDropdownV2Beta,
  GuxListbox,
  GuxOptionV2
];
const language = 'en';

describe('gux-pagination-items-per-page', () => {
  describe('#render', () => {
    [
      { itemsPerPage: 25 },
      { itemsPerPage: 50 },
      { itemsPerPage: 75 },
      { itemsPerPage: 100 }
    ].forEach(({ itemsPerPage }, index) => {
      it(`should render as expected (${index + 1})`, async () => {
        const html = `
          <gux-pagination-items-per-page
            items-per-page="${itemsPerPage}">
          </gux-pagination-items-per-page>
        `;
        const page = await newSpecPage({ components, html, language });

        expect(page.rootInstance).toBeInstanceOf(GuxPaginationItemsPerPage);

        expect(page.root).toMatchSnapshot();
      });
    });
  });
});
