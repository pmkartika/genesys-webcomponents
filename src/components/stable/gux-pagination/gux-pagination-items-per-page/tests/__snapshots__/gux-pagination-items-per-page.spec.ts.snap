// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gux-pagination-items-per-page #render should render as expected (1) 1`] = `
<gux-pagination-items-per-page items-per-page="25">
  <div class="gux-pagination-items-per-page-container">
    <div class="gux-pagination-items-per-page-picker">
      <gux-dropdown-v2-beta>
        <mock:shadow-root>
          <gux-popup-beta>
            <div class="gux-target-container-collapsed" slot="target">
              <button aria-expanded="false" aria-haspopup="listbox" class="gux-field gux-field-button" type="button">
                <div class="gux-field-content">
                  <div class="gux-selected-option">
                    25
                  </div>
                </div>
                <gux-icon class="gux-expand-icon" iconname="chevron-small-down" screenreader-text="Dropdown"></gux-icon>
              </button>
            </div>
            <div class="gux-listbox-container" slot="popup">
              <slot></slot>
            </div>
          </gux-popup-beta>
        </mock:shadow-root>
        <gux-listbox aria-label="Items per page" role="listbox" tabindex="0">
          <mock:shadow-root>
            <slot></slot>
          </mock:shadow-root>
          <gux-option-v2 aria-disabled="false" aria-selected="true" class="gux-selected" id="gux-option-v2-i" role="option">
            <!---->
            25
          </gux-option-v2>
          <gux-option-v2 aria-disabled="false" aria-selected="false" id="gux-option-v2-i" role="option">
            <!---->
            50
          </gux-option-v2>
          <gux-option-v2 aria-disabled="false" aria-selected="false" id="gux-option-v2-i" role="option">
            <!---->
            75
          </gux-option-v2>
          <gux-option-v2 aria-disabled="false" aria-selected="false" id="gux-option-v2-i" role="option">
            <!---->
            100
          </gux-option-v2>
        </gux-listbox>
      </gux-dropdown-v2-beta>
    </div>
    <div>
      per page
    </div>
  </div>
</gux-pagination-items-per-page>
`;

exports[`gux-pagination-items-per-page #render should render as expected (2) 1`] = `
<gux-pagination-items-per-page items-per-page="50">
  <div class="gux-pagination-items-per-page-container">
    <div class="gux-pagination-items-per-page-picker">
      <gux-dropdown-v2-beta>
        <mock:shadow-root>
          <gux-popup-beta>
            <div class="gux-target-container-collapsed" slot="target">
              <button aria-expanded="false" aria-haspopup="listbox" class="gux-field gux-field-button" type="button">
                <div class="gux-field-content">
                  <div class="gux-selected-option">
                    50
                  </div>
                </div>
                <gux-icon class="gux-expand-icon" iconname="chevron-small-down" screenreader-text="Dropdown"></gux-icon>
              </button>
            </div>
            <div class="gux-listbox-container" slot="popup">
              <slot></slot>
            </div>
          </gux-popup-beta>
        </mock:shadow-root>
        <gux-listbox aria-label="Items per page" role="listbox" tabindex="0">
          <mock:shadow-root>
            <slot></slot>
          </mock:shadow-root>
          <gux-option-v2 aria-disabled="false" aria-selected="false" id="gux-option-v2-i" role="option">
            <!---->
            25
          </gux-option-v2>
          <gux-option-v2 aria-disabled="false" aria-selected="true" class="gux-selected" id="gux-option-v2-i" role="option">
            <!---->
            50
          </gux-option-v2>
          <gux-option-v2 aria-disabled="false" aria-selected="false" id="gux-option-v2-i" role="option">
            <!---->
            75
          </gux-option-v2>
          <gux-option-v2 aria-disabled="false" aria-selected="false" id="gux-option-v2-i" role="option">
            <!---->
            100
          </gux-option-v2>
        </gux-listbox>
      </gux-dropdown-v2-beta>
    </div>
    <div>
      per page
    </div>
  </div>
</gux-pagination-items-per-page>
`;

exports[`gux-pagination-items-per-page #render should render as expected (3) 1`] = `
<gux-pagination-items-per-page items-per-page="75">
  <div class="gux-pagination-items-per-page-container">
    <div class="gux-pagination-items-per-page-picker">
      <gux-dropdown-v2-beta>
        <mock:shadow-root>
          <gux-popup-beta>
            <div class="gux-target-container-collapsed" slot="target">
              <button aria-expanded="false" aria-haspopup="listbox" class="gux-field gux-field-button" type="button">
                <div class="gux-field-content">
                  <div class="gux-selected-option">
                    75
                  </div>
                </div>
                <gux-icon class="gux-expand-icon" iconname="chevron-small-down" screenreader-text="Dropdown"></gux-icon>
              </button>
            </div>
            <div class="gux-listbox-container" slot="popup">
              <slot></slot>
            </div>
          </gux-popup-beta>
        </mock:shadow-root>
        <gux-listbox aria-label="Items per page" role="listbox" tabindex="0">
          <mock:shadow-root>
            <slot></slot>
          </mock:shadow-root>
          <gux-option-v2 aria-disabled="false" aria-selected="false" id="gux-option-v2-i" role="option">
            <!---->
            25
          </gux-option-v2>
          <gux-option-v2 aria-disabled="false" aria-selected="false" id="gux-option-v2-i" role="option">
            <!---->
            50
          </gux-option-v2>
          <gux-option-v2 aria-disabled="false" aria-selected="true" class="gux-selected" id="gux-option-v2-i" role="option">
            <!---->
            75
          </gux-option-v2>
          <gux-option-v2 aria-disabled="false" aria-selected="false" id="gux-option-v2-i" role="option">
            <!---->
            100
          </gux-option-v2>
        </gux-listbox>
      </gux-dropdown-v2-beta>
    </div>
    <div>
      per page
    </div>
  </div>
</gux-pagination-items-per-page>
`;

exports[`gux-pagination-items-per-page #render should render as expected (4) 1`] = `
<gux-pagination-items-per-page items-per-page="100">
  <div class="gux-pagination-items-per-page-container">
    <div class="gux-pagination-items-per-page-picker">
      <gux-dropdown-v2-beta>
        <mock:shadow-root>
          <gux-popup-beta>
            <div class="gux-target-container-collapsed" slot="target">
              <button aria-expanded="false" aria-haspopup="listbox" class="gux-field gux-field-button" type="button">
                <div class="gux-field-content">
                  <div class="gux-selected-option">
                    100
                  </div>
                </div>
                <gux-icon class="gux-expand-icon" iconname="chevron-small-down" screenreader-text="Dropdown"></gux-icon>
              </button>
            </div>
            <div class="gux-listbox-container" slot="popup">
              <slot></slot>
            </div>
          </gux-popup-beta>
        </mock:shadow-root>
        <gux-listbox aria-label="Items per page" role="listbox" tabindex="0">
          <mock:shadow-root>
            <slot></slot>
          </mock:shadow-root>
          <gux-option-v2 aria-disabled="false" aria-selected="false" id="gux-option-v2-i" role="option">
            <!---->
            25
          </gux-option-v2>
          <gux-option-v2 aria-disabled="false" aria-selected="false" id="gux-option-v2-i" role="option">
            <!---->
            50
          </gux-option-v2>
          <gux-option-v2 aria-disabled="false" aria-selected="false" id="gux-option-v2-i" role="option">
            <!---->
            75
          </gux-option-v2>
          <gux-option-v2 aria-disabled="false" aria-selected="true" class="gux-selected" id="gux-option-v2-i" role="option">
            <!---->
            100
          </gux-option-v2>
        </gux-listbox>
      </gux-dropdown-v2-beta>
    </div>
    <div>
      per page
    </div>
  </div>
</gux-pagination-items-per-page>
`;
